client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/HashBasedShuffleWriter.java

```diff
@@ -273,7 +273,7 @@ private void pushGiantRecord(int partitionId, byte[] buffer, int numBytes) throw
         shuffleClient.pushData(
             shuffleId,
             mapId,
-            taskContext.attemptNumber(),
+            SparkUtils.getMapAttemptNumber(taskContext),
```

Comment:
Instead of `SparkUtils`, which is version specific, create a util in common and add `getMapAttemptNumber` there ?
This is going to be the same across spark versions.